test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1444,27 +1444,51 @@ UCS_TEST_SKIP_COND_P(test_ucp_sockaddr_wireup, compare_cm_and_wireup_configs,
     }
 }
 
-class test_max_lanes : public test_ucp_sockaddr {
+template<size_t NumLanes> class test_max_lanes : public test_ucp_sockaddr {
 public:
     static void get_test_variants(std::vector<ucp_test_variant> &variants)
     {
         get_test_variants_cm_mode(variants, UCP_FEATURE_TAG, CONN_REQ_TAG,
                                   "tag");
     }
+
+    void init() override
+    {
+        auto num_lanes_str = ucs::to_string(NumLanes);
+        modify_config("MAX_RNDV_RAILS", num_lanes_str); // for protov1
```

Comment:
maybe just check is_proto_enabled() here and set num_lanes_str accordingly? instead of using class template

Reply:
That's not the only difference, 16 lanes test is executed with IB transport which involves DC lanes creation. I think it worth to test that separately.

Reply:
why not instantiate the test on both rc and dc?

Reply:
For 16 lanes it is possible. For 64 lanes there is no enough DCI/ DCI pools to create 64 lanes.

Reply:
[That](https://github.com/openucx/ucx/blob/748033c40c58828a92d76c860763ebdc1058a2d1/src/uct/ib/dc/dc_mlx5.c#L1511) error happens and as I understand that's not a bug, but just current limitation for DC. So we still can create up to 16 DC lanes.

Reply:
so maybe check if DC is in the list of transports, and if yes limit max_lanes to 16, same as we're doing for protov1?

Reply:
Done!